### PR TITLE
Use redux

### PR DESCRIPTION
**Pull request by digia3d**
In this part of the project, the assignment was to:

-  implement Redux in our React App.
-  To import Redux Provider and store in the main component.
-  To use a selector to consume the state.
-  Use dispatch to import Redux.
-  Generate a unique id and add them to an object.
-  Add functionality for the "Remove" button.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,28 +1,67 @@
...
 
 function reducer(state = initialState, action) {
   switch (action.type) {
     case ADD_BOOK:
-    default: return state.books;
+      return [...state, action.payload];
+
+    case REMOVE_BOOK:
+      return [state.filter((book) => book.id !== action.payload.id)];
+
```
Comment (omarsalem7) [inline src/redux/books/books.js]: - [x] Kindly check remove k=logic for removing book, there is a strange behaviour when removing it, you can check [this video](https://www.loom.com/share/0ace5020c928452bb561896ef1e6abd4) for what I mean.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @digia3d 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Descriptive pull request ✔️
- No linter errors ✔️
- Descriptive commit messages ✔️

## Required Changes ♻️

- [x]  _Check the comments under the review._

- [x] Add functionality for check status in categories page.

image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/pages/Categories.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
+  const status = useSelector((state) => state.checkStatus);
+  const dispatch = useDispatch();
 
-function Categories() {
   return (
     <>
       <div>
-        <button type="button">Check status</button>
+        <button type="button" onClick={() => dispatch(checkStatus())}>Check status</button>
+        <p>{status}</p>
```
Comment (Ugboaja-Uchechi) [inline src/components/pages/Categories.js:13]: - [ ] {OPTIONAL] Kindly make sure when a user clicks on the Check Status button it displays "Under Construction" 🙏

**Review by Ugboaja-Uchechi — APPROVED** [review]
Hi @digia3d 

## Status: APPROVED 🥳 🥳 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- Linters are passing ✔️ 
- README is descriptive 👍 
- Great job with your design and functionalities 👏 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**